src/DeltaLake/DeltaTable.cs

```diff
@@ -337,9 +337,12 @@ public Builder Add(IEnumerable<RecordBatch> data, Action<AddOptions>? configure
             using var options = new AddOptions();
             configure?.Invoke(options);
             using var arrowProperties = options.ArrowProperties.Build();
+
+            Compression compression = options.Compression.ToParquetSharpCompression();
+            options.ParquetProperties.Compression(compression);
```

Comment:
Could not see that compression from the action func where actually used here, so added this.
Perhaps I'm missing something

Reply:
Nice catch :D